devito/symbolics/inspection.py

```diff
@@ -6,7 +6,26 @@
 from devito.logger import warning
 from devito.tools import as_tuple, flatten
 
-__all__ = ['count', 'estimate_cost']
+__all__ = ['compare_ops', 'count', 'estimate_cost']
+
+
+def compare_ops(e1, e2):
+    """
+    Return True if the two expressions ``e1`` and ``e2`` perform the same arithmetic
+    operations over the same input operands, False otherwise.
+    """
+    if type(e1) == type(e2) and len(e1.args) == len(e2.args):
```

Comment:
I'm surprised sympy doesn't have an Expression comparison we can't reuse, but looks fine.

Reply:
it does (though it's buried)

However, this one ignores the indices of an Indexed , ie it compares down up to symbol/indexed labels

```
In [11]: compare_ops(u[t, x, y] + u[t, x+1, y+1], u[t, x, y] + u[t, x+1, y-1])                                                                                                                                     
Out[11]: True

In [12]: compare_ops(u[t, x, y] + u[t, x+1, y+1], u[t, x, y] - u[t, x+1, y-1])                                                                                                                                     
Out[12]: False

In [13]: compare_ops(u[t, x, y] + u[t, x+1, y+1], u[t, x, y] * u[t, x+1, y-1])                                                                                                                                     
Out[13]: False

In [15]: compare_ops(u[t, x, y] + u[t, x+1, y+1], u[t, x, y] + v[t, x+1, y+1])                                                                                                                                     
Out[15]: False
```

I'm updating the docstring now. You're right, it's misleading.